compiler/include/type.h

```diff
@@ -364,6 +364,29 @@ class PrimitiveType : public Type {
 };
 
 
+/************************************* | **************************************
+*                                                                             *
+* a ConstrainedType is the type of:                                           *
+* (1) a formal of a CG function if it is subject to interface constraint(s)   *
+* (2) a formal of an InterfaceSymbol if it has the type intent                *
```

Comment:
Are there any other formals to InterfaceSymbols? Can you have this comment show syntax examples?